### PR TITLE
Project: Members only

### PR DESCRIPTION
**Pull request by rna**
In this project:
- Created User and Post data models
- Provided authentication for the User to Login
- Added User remember feature to browser using `Cookies.permanent` 
- Created a logic so that only members can write the post.
- Added basic UI with user Registration page, Login page & Post Index page
- Created a view logic so that only members can view the author of a Post

### CODE REVIEW

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+module SessionsHelper
+  def log_in(user)
+    session[:user_id] = user.id
```
Comment (maelfosso) [inline app/helpers/sessions_helper.rb]: No `session` please

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  end
+
+  def remember(user)
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
```
Comment (maelfosso) [inline app/helpers/sessions_helper.rb]: It's not over, you should set the `current_user`

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
+        if user && user.authenticated?(cookies[:remember_token])
+          log_in user
+          @current_user = user
+        end
+    end
+  end
```
Comment (maelfosso) [inline app/helpers/sessions_helper.rb]: No `session`, please.
Read this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/current_user_requirement.png), you will see that you can write it in one line

File: app/helpers/sessions_helper.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  def forget(user)
+    user.forget
+    cookies.delete(:user_id)
+    cookies.delete(:remember_token)
+  end
+
+  def log_out
+    forget(current_user)
+    session.delete(:user_id)
```
Comment (maelfosso) [inline app/helpers/sessions_helper.rb]: No `session`

**Comment by rna** [comment]
Hi @maelfosso 

Thanks for your time and feedback.

All the necessary changes are done.

Thanks

**Comment by rna** [comment]
Hi @maelfosso 

Thanks for your time and feedback.

All the necessary changes are done. Please review.

Thanks

**Comment by rna** [comment]
Hey @Torres-ssf 

Thanks for your time and feedback.
You are right. I almost forgot the most important aspects. Thanks for reminding. 

All the above comments addressed.  (i.e. PR Summary updated & stickler added)

Please check and confirm.

Thanks

**Pull request merged**